Mathlib/Order/RelSeries.lean

```diff
@@ -521,6 +648,42 @@ lemma eraseLast_last_rel_last (p : RelSeries r) (h : p.length ≠ 0) :
   convert p.step ⟨p.length - 1, by omega⟩
   simp only [Nat.succ_eq_add_one, Fin.succ_mk]; omega
 
+lemma snoc_self_eraseLast (p : RelSeries r) (h : p.length ≠ 0) :
+    p.eraseLast.snoc p.last (p.eraseLast_last_rel_last h) = p := by
+  symm
+  have len : p.length - 1 + 1 = p.length := Nat.succ_pred_eq_of_ne_zero h
+  ext n
+  · simpa using len.symm
+  by_cases eq : n.1 < p.length - 1 + 1
+  · simp [snoc, append, Fin.append, Fin.addCases, eq]
+  · have : n = p.length := by
+      rw [len] at eq
+      exact Fin.eq_of_val_eq (by simp [Nat.eq_of_lt_succ_of_not_lt n.2 eq])
+    simp [snoc, append, Fin.append, Fin.addCases, eq, this, last]
```

Comment:
Similarly, this might be better done with a lemma along the lines of:
```lean
@[simp] lemma toList_eraseLast (p : RelSeries r) :
    p.eraseLast.toList = p.toList.dropLast := sorry
```

and using `toList_injective` plus `List.dropLast_append_getLast`.